freqtrade/optimize/hyperopt_auto.py

```
@@ -0,0 +1,83 @@
+"""
+HyperOptAuto class.
+This module implements a convenience auto-hyperopt class, which can be used together with strategies that implement
+IHyperStrategy interface.
+"""
+from typing import Any, Callable, Dict, List
+from pandas import DataFrame
+from skopt.space import Categorical, Dimension, Integer, Real  # noqa
+
+from freqtrade.optimize.hyperopt_interface import IHyperOpt
+
+
+# noinspection PyUnresolvedReferences
+class HyperOptAuto(IHyperOpt):
+    """
+    This class delegates functionality to Strategy(IHyperStrategy) and Strategy.HyperOpt classes. Most of the time
+    Strategy.HyperOpt class would only implement indicator_space and sell_indicator_space methods, but other hyperopt
+    methods can be overridden as well.
+    """
+    def buy_strategy_generator(self, params: Dict[str, Any]) -> Callable:
+        assert hasattr(self.strategy, 'enumerate_parameters'), 'Strategy must inherit from IHyperStrategy.'
```

you'll need to be careful with using assert statements in non-test code.

I know it's a (kindof) simple/quick way to fail if something doesn't happen - however, assert failing can be disabled on a python level (`python -O ...` ) - which would have all assert-statements silently pass, and should therefore not be used in final / production code.

Therefore freqtrade usually uses "OperationalException" to fail - as we handle that gracefully before exiting (i know that'll convert this oneline into 2 lines... ).

---

Will do! 👍🏻